README.md

```diff
@@ -71,6 +71,28 @@ spec:
   enabled: false
 ```
 
+### Upload kubeconfig to GitHub secrets
+
+In order to upload Kubeconfigs to GitHub you need to start klum with a valid GitHub token `--github-token` and add the following annotations to your User.
+`github/env` can be skipped if you want only repository secrets.  
+`github/name` can be skipped and a name will be generated from the name of the user and the postfix `_KUBECONFIG`. In this case: `DARREN_KUBECONFIG`. 
+
+```yaml
+kind: User
+apiVersion: klum.cattle.io/v1alpha1
+metadata:
+  name: darren
+  annotations:
+    "github/owner": jadolg
+    "github/repo": klum-example
+    "github/env": prod
+    "github/name": KUBE_CONFIG
+spec:
+  roles:
+    - namespace: default
+      clusterRole: cluster-admin
```

Comment:
Since this CRD is provided by Klum, I suggest updating that instead of using untyped annotations.

Such as:

```yaml
kind: User
apiVersion: klum.cattle.io/v1alpha1
metadata:
  name: darren
spec:
  roles:
    - namespace: default
      clusterRole: cluster-admin
  sync:
    github:
      organizationSecrets:
        - name: KUBE_CONFIG
          owner: jadolg
          credentials: my-github-pat

      repositorySecrets:
        - name: KUBE_CONFIG
          owner: jadolg
          repository: klum-example
          credentials: my-github-app

      environmentSecrets:
        - name: KUBE_CONFIG
          owner: jadolg
          repository: klum-example
          environment: prod
          credentials: my-github-app

    onepassword:
      items:
        - name: kubeconfig for my test cluster
          vaultName: kubeconfigs
          credentials: my-onepassword-connect

---
apiVersion: v1
kind: Secret
metadata:
  name: my-klum-config
stringData:
  config: |
    credentials:
      github:
        - name: my-github-pat
          type: PAT
          github-api: https://api.github.com
          token: token-goes-here

        - name: my-github-app
          type: App
          github-api: https://api.github.com
          app-id: "123"
          app-installation-id: "456"
          app-key: |
            -----BEGIN RSA PRIVATE KEY-----
            ...
            -----END RSA PRIVATE KEY-----

      onepassword-connect:
        - name: my-onepassword-connect
          url: https://1p-connect.example.com
          token: token-goes-here
```

Maybe `sync` is a weird naming. But you get what I'm going for?

Also think it'd be wise to not have 1 GitHub credential globally, but instead select which credential to use for each sync integration

Reply:
I think we can iterate until we get to this point.
I'd like to keep the one credential approach for now not to overcomplicate the solution.

Reply:
Sure! I'd suggest still going with CRD changes though, but yeah you can skip the `credentials` field and config file

Reply:
Also, I'd like to keep Org out, at least for now.
It's better to have several secrets which we can anyways easily manage here.